test/e2e/all_in_one_test.go

```diff
@@ -377,3 +379,51 @@ func TestDeployAllInOneEnterprisePostgres(t *testing.T) {
 	verifyEnterprise(ctx, t, env, adminPassword)
 	verifyEnterpriseWithPostgres(ctx, t, env, adminPassword)
 }
+
+func TestDeployAllInOneDBLESSMultiGW(t *testing.T) {
+	t.Parallel()
+
+	const (
+		manifestFileName = "all-in-one-dbless-multi-gw.yaml"
+		manifestFilePath = "../../deploy/single/" + manifestFileName
+	)
+
+	t.Logf("configuring %s manifest test", manifestFileName)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	builder, err := getEnvironmentBuilder(ctx)
+	require.NoError(t, err)
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+
+	defer func() {
+		helpers.TeardownCluster(ctx, t, env.Cluster())
+	}()
+
+	t.Log("deploying kong components")
+	f, err := os.Open(manifestFilePath)
+	require.NoError(t, err)
+	defer f.Close()
+	var manifest io.Reader = f
+
+	manifest, err = patchControllerImageHelper(manifest, manifestFilePath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+
+	t.Log("running ingress tests to verify all-in-one deployed ingress controller and proxy are functional")
+	deployIngress(ctx, t, env)
+	verifyIngress(ctx, t, env)
+
+	gatewayDeployment, err := env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Get(ctx, "proxy-kong", metav1.GetOptions{})
+	require.NoError(t, err)
+	gatewayDeployment.Spec.Replicas = lo.ToPtr(int32(3))
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Update(ctx, gatewayDeployment, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Log("confirming that routes are stil served after scaling out")
+	verifyIngress(ctx, t, env)
+	verifyIngress(ctx, t, env)
+	verifyIngress(ctx, t, env)
```

Comment:
Can we somehow explicitly verify that all proxy pods have the same configuration (e.g. with `GET /config`)?